authentication/views/auth.py

```diff
@@ -92,10 +106,80 @@ def activation_view(request, code: str):
             except:
                 form.add_error(None, "The code or email provided was invalid.")
     else:
-        form = ActivationForm(None)
+        form = CustomSetPasswordForm(None)
     return render(request, "authentication/activation.html", {"form": form})
 
 
+def forgot_password_view(request):
+    if request.method == "POST":
+        form = ForgotPasswordForm(data=request.POST)
+        if form.is_valid():
+            results = User.objects.filter(email=form.cleaned_data["email"])
+            if results:
+                user = results[0]
+                signer = TimestampSigner()
+                signed_value = signer.sign(user.email)
+                code = signed_value[signed_value.find(":") + 1 :]
+
+                forgot_password_record = ForgotPassword(user=user, code=code)
+                forgot_password_record.save()
+
+                messages.success(
+                    request,
+                    "If an account exists with the email address a reset password will be emailed to you. Please check your email and follow instructions from there.",
+                )
+
+                send_forgot_password_email(request, user)
+
+        else:
+            form.add_error("email", "Invalid email address.")
+    else:
+        form = ForgotPasswordForm()
+    return render(request, "authentication/forgot_password.html", {"form": form})
+
+
+def reset_password_view(request, code: str):
+    if request.method == "POST":
+        form = CustomSetPasswordForm(user=None, data=request.POST)
+        forgot_password_record = get_object_or_404(ForgotPassword, code=code)
+        form.user = forgot_password_record.user
+        code_is_valid(
+            code,
+            request.POST["email"],
+            forgot_password_record.user,
+            settings.RESET_PASSWORD_EXPIRATION_DAYS,
+        )
```

Comment:
Ditto previous comment - I don't see an evaluation of this function result (True/False).